tests/integration/components/bookmark-form/component-test.js

```diff
@@ -0,0 +1,23 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('bookmark-form', 'Integration | Component | bookmark form', {
+  integration: true
+});
+
+test('it sets focus on the title field when empty', function(assert) {
+  this.set('bookmark', {title: null});
+  this.render(hbs`{{bookmark-form bookmark=bookmark}}`);
```

Comment:
The `this.set` just sets the outer context, while the `this.render` has the way the component would actually be called.

E.g. you could also have something like

```javascript
this.set('model', {bookmark: { title: null}});
this.render(hbs`{{bookmark-form bookmark=model.bookmark}}`);
```

I hope this makes the difference clearer.